### PR TITLE
Modify docstrings for consistency

### LINKED ISSUE #171: Google style docstrings throughout CKT
All docstrings should follow the same pattern and have consistent styling.

We may want to follow up on this and begin to create a style guide for this repo.

### PR DESCRIPTION
**Pull request by caleb-johnson**
Resolves #171 

Refer to:
https://google.github.io/styleguide/pyguide.html#doc-function-args

This PR aims to standardize all docstrings in CKT. This will make things easier for new users going through all of our docs, and it will also make our docstrings less verbose and more readable.

The main things changed throughout all of these files are:
 - No tick marks for arg lists
 - No redundant type hints in docstrings (everything is type hinted in the source)
 - Capitalize first letter of each arg/return/raises explanation
 - All raises items will begin on the same line as the exception thrown
 - Return explanations will be plain language explanations of the return item.

### CODE REVIEW

File: circuit_knitting_toolbox/circuit_cutting/cutqc/mip_model.py [hunk 1]
```diff
@@ -425,15 +396,14 @@ def pwl_exp(
...
+            lb: lower bound
+            ub: upper bound
+            base: the base of the input exponential
+            coefficient: the coefficient of the original exponential
+            integer_only: whether the input x's are only integers
 
         Returns:
-            - (list): the x's of the piecewise approximation
-            - (list): the f(x)'s of the piecewise approximation
+            A tuple containing the x's and f(x)'s of the piecewise approximation
```
Comment (garrison) [inline circuit_knitting_toolbox/circuit_cutting/cutqc/mip_model.py:406]:
```suggestion
            A tuple containing the :math:`x`\ s and :math:`f(x)`\ s of the piecewise approximation
```

File: circuit_knitting_toolbox/circuit_cutting/qpd/qpd_basis.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def from_gate(gate: Gate) -> "QPDBasis":
         calls :func:`~qpd.qpd.qpdbasis_to_gate` under the hood.
 
         Returns:
-            - (QPDBasis): The newly-instantiated QPDBasis object
+            The newly-instantiated QPDBasis object
```
Comment (garrison) [inline circuit_knitting_toolbox/circuit_cutting/qpd/qpd_basis.py:141]:
```suggestion
            The newly-instantiated :class:`QPDBasis` object
```

File: circuit_knitting_toolbox/circuit_cutting/qpd/qpd_basis.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def probabilities(self) -> Sequence[float]:
         Get the probabilities on which the maps will be sampled.
 
         Returns:
-            - (Sequence[float]): The probabilities member variable
+            The probabilities member variable
```
Comment (garrison) [inline circuit_knitting_toolbox/circuit_cutting/qpd/qpd_basis.py:104]:
```suggestion
            The ``probabilities`` member variable
```

File: circuit_knitting_toolbox/circuit_cutting/qpd/qpd_basis.py [hunk 4]
```diff
@@ -82,7 +82,7 @@ def coeffs(self) -> Sequence[float]:
         Quasiprobability decomposition coefficients.
 
         Returns:
-            - (Sequence[float]): The coeffs member variable
+            The coeffs member variable
```
Comment (garrison) [inline circuit_knitting_toolbox/circuit_cutting/qpd/qpd_basis.py:85]:
```suggestion
            The ``coeffs`` member variable
```

**Review by garrison — APPROVED** [review]
Thank you!

I made a few comments that I noticed, but this looks great to me.  I assume there are probably some stragglers somewhere (but maybe you got them all).  Even so, I'd rather just get this merged rather than take more time to evaluate every single last change that might need to happen beyond this.  Since this touches so many lines, I don't want it to get far out of touch with the `main` branch.  I added a few suggestions that I noticed while looking this over, but that's all.

**Comment by caleb-johnson** [comment]
@garrison I didnt mean to ping you for review, I feel like it happened automatically since I didnt make it a draft PR.

Anyway, this is done now, with the exception of addressing your peer review. I will hold off on addressing it until you finish your review of all the new stuff I just pushed

Thanks for the quick review!

**Pull request merged**